### PR TITLE
#157658365 Setup testing framework

### PR DESCRIPTION
**Pull request by chukwuemekachm**
#### What does this PR do?
Sets up testing framework using mocha
#### Description of Task to be completed?
Install mocha chai and chai-http as dev dependencies, write test for server
#### How should this be manually tested?
On the console type npm test
#### What are the relevant pivotal tracker stories?
#157658365
#### Screenshots (if appropriate)

screenshot 128
screenshot 126

### CODE REVIEW

File: Server/tests/server.test.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+/** imports */
+import chai from 'chai';
+import chaiHttp from 'chai-http';
+import { server } from '../../app';
+
+/** instantiations */
+const should = chai.should();
+chai.use(chaiHttp);
+
+/** tests */
```
Comment (paradisekelechi) [inline Server/tests/server.test.js]: Comments should tell us something meaningful

Comment (paradisekelechi) [inline Server/tests/server.test.js]: No need for this comment

File: Server/tests/server.test.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+describe('GET /', () => {
```
Comment (paradisekelechi) [inline Server/tests/server.test.js]: This does all the description. No need for comments

File: package.json [hunk 3]
```diff
@@ -4,7 +4,7 @@
     "description": "Maintenance Tracker App is an application that provides users with the ability to reach out to operations or repairs department regarding repair or maintenance requests and monitor the status of their request.",
     "main": "app.js",
     "scripts": {
-        "test": "echo \"Error: no test specified\" && exit 1",
+        "test": "mocha --exit --require babel-register ./Server/tests/**/*.test.js",
         "start": "nodemon --exec babel-node app.js"
```
Comment (paradisekelechi) [inline package.json]: Do not use nodemon in the start script but you can use it in the start:dev script

File: Server/tests/server.test.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    chai.request(server)
+      .get('/')
+      .end((req, res) => {
+        res.should.have.status(200);
+        res.body.should.have.property('message');
+        done();
+      });
+  });
+
+  it('should return 404 and message, "Route is not supported on the server." when an unavailable route is requested', (done) => {
```
Comment (paradisekelechi) [inline Server/tests/server.test.js]: This description is too much. Just mention what this tests.

File: Server/tests/server.test.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    chai.request(server).get('/api').end((req, res) => {
+      res.should.have.status(404);
```
Comment (paradisekelechi) [inline Server/tests/server.test.js]: Is it only the status that is returned? Test everything you can test eg message, array, objects, etc

**Comment by chukwuemekachm** [comment]
I have resolved them now

**Pull request merged**